lib/discourse.rb

```diff
@@ -1116,16 +1116,7 @@ def self.preload_rails!
       end
     end
 
-    schema_cache = ActiveRecord::Base.connection.schema_cache
-
     RailsMultisite::ConnectionManagement.safe_each_connection do
-      # load up schema cache for all multisite assuming all dbs have
-      # an identical schema
-      dup_cache = schema_cache.dup
-      # this line is not really needed, but just in case the
-      # underlying implementation changes lets give it a shot
-      dup_cache.connection = nil
-      ActiveRecord::Base.connection.schema_cache = dup_cache
```

Comment:
I am curious about the reason for making this change. Is it because the Rails API is no longer supported? How does it affect preloading if this change is removed?

Reply:
Indeed, you can’t set the schema cache on the connection anymore. There’s a place where it might be feasible, but it’s buried relatively deep (on the pool config object) and that method has been removed from the current Rails code. From what I understand, though, the schema cache is automatically created per connection once. So I guess the worst-case scenario is that it will be fetched when the first connection is established (for each connection) instead of sharing the same one here like we were doing. I would tend to say it’s probably not that impactful, but I can’t really give you a definitive answer here.